SourceCode/include/tree/KDTree.h

```diff
@@ -0,0 +1,198 @@
+#include <algorithm>
+#include <cstddef>
+#include <limits>
+#include <numeric>
+#include <optional>
+#include <stack>
+#include <vector>
+
+#include "tracer/Ray.h"
+#include "tracer/Scene.h"
+#include "tracer/Triangle.h"
+#include "tracer/Vector.h"
+
+class BoundingBox {
+ private:
+  Vector minPoint;
+  Vector maxPoint;
+
+ public:
+  BoundingBox(const Vector &minPoint, const Vector &maxPoint) : minPoint{minPoint}, maxPoint{maxPoint} {}
+  explicit BoundingBox(const Scene &scene) {
+    for (auto &triangle : scene.triangles) {
```

Comment:
You could construct this box during the parsing of the scene instead of iterating all triangles again here